cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -0,0 +1,96 @@
+# Various Ruby helper methods to generate Varnish VCL from the base configuration.
+
+# Basic regex matcher for the query part of a URL.
+QUERY_REGEX = "(\\?[a-z0-9]+)?$"
+
+def path_to_regex(path)
+  path = "^/#{path.sub(/^\//,'')}"
+  if path[-1] == '*'
+    path.slice!(-1)
+  else
+    path = path + QUERY_REGEX
+  end
+  path
+end
+
+# Varnish uses 'bereq' in the 'response' section, 'req' otherwise.
+def req(section)
+  section == 'response' ? 'bereq' : 'req'
+end
+
+# Returns a regex-matcher string based on an array of filename extensions.
+def extensions_to_regex(exts)
+  return [] if exts.empty?
+  ["(?i)(#{exts.join('|').gsub('.','\.')})#{QUERY_REGEX}"]
+end
+
+# Returns a regex-conditional string fragment based on the provided behavior.
+# In the 'proxy' section, ignore extension-based behaviors (e.g., *.png).
+def paths_to_regex(path_config, section)
```

Comment:
Please document the valid path patterns and add unit tests that verify that allowed patterns are correctly translated and that disallowed patterns are rejected with an error.  The CloudFront document mentioned earlier in this file allows '*' and '?' anywhere in the path, but this appears to be more restrictive in what it can handle.